src/coreclr/tools/Common/Compiler/DependencyAnalysis/ShadowConcreteMethodNode.cs

```diff
@@ -85,14 +85,37 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
             return dependencies;
         }
 
+        public sealed override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory factory)
+        {
+            // Instantiate the runtime determined dependencies of the canonical method body
+            // with the concrete instantiation of the method to get concrete dependencies.
+            Instantiation typeInst = Method.OwningType.Instantiation;
+            Instantiation methodInst = Method.Instantiation;
+            IEnumerable<CombinedDependencyListEntry> staticDependencies = CanonicalMethodNode.GetConditionalStaticDependencies(factory);
+
+            if (staticDependencies != null)
+            {
+                foreach (CombinedDependencyListEntry canonDep in staticDependencies)
+                {
+                    Debug.Assert(canonDep.OtherReasonNode is not INodeWithRuntimeDeterminedDependencies);
+
+                    var node = canonDep.Node;
+                    if (node is INodeWithRuntimeDeterminedDependencies runtimeDeterminedNode)
```

Comment:
I think https://github.com/dotnet/runtime/pull/103883#discussion_r1669800678 explains it, but skips a small step. I wrote "Canonical method `Blah` depends on e.g. T__Canon (for the `typeof(T)`).", but the actual dependency is: "Canonical method `Blah` depends on ReadyToRunGenericHelperNode of T__Canon."

So this is how directly instantiated things are tracked. The remaining MakeGenericMethodSite and MakeGenericTypeSite are relatively recent additions from #99037 and are only used in relation to dataflow since the problem is similar (we compute something on a "lesser instantiated thing" and need to specialize it for whatever else is in the dependency graph).